src/script.js

```diff
@@ -151,9 +151,9 @@ function decompilePushOnly (script) {
 
   return chunks.map(function (op) {
     if (Buffer.isBuffer(op)) return op
-    if (op === OPS.OP_0) return new Buffer(0)
+    if (op === OPS.OP_0) return op
 
-    return scriptNumber.encode(op - OP_INT_BASE)
+    return op
   })
```

Comment:
https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/src/transaction.js#L449 - maybe we should just be putting the script number encoding code there?